cmd/admin-rpc-server.go

```diff
@@ -57,11 +58,50 @@ func (s *adminCmd) Restart(args *AuthRPCArgs, reply *AuthRPCReply) error {
 
 // ListLocks - lists locks held by requests handled by this server instance.
 func (s *adminCmd) ListLocks(query *ListLocksQuery, reply *ListLocksReply) error {
+	if err := query.IsAuthenticated(); err != nil {
+		return err
+	}
 	volLocks := listLocksInfo(query.bucket, query.prefix, query.relTime)
 	*reply = ListLocksReply{volLocks: volLocks}
 	return nil
 }
 
+func (s *adminCmd) ReInitDisks(args *AuthRPCArgs, reply *AuthRPCReply) error {
+	if err := args.IsAuthenticated(); err != nil {
+		return err
+	}
+
+	// Get the current object layer instance.
+	objLayer := newObjectLayerFn()
+
+	if _, ok := objLayer.(*xlObjects); !ok {
+		return errors.New("not supported for non erasure-code backend")
+	}
+	// Initialize new disks to include the newly formatted disks.
+	bootstrapDisks, err := initStorageDisks(globalEndpoints)
+	if err != nil {
+		return err
+	}
+
+	// Initialize new object layer with newly formatted disks.
+	newObjectAPI, err := newXLObjects(bootstrapDisks)
+	if err != nil {
+		return err
+	}
+
+	// Replace object layer with newly formatted storage.
+	globalObjLayerMutex.Lock()
+	{
```

Comment:
Is this kind of scoping necessary? - are you using Go's behavior specifically here?.  If yes mention that why this is done so to avoid someone removing this.